app/controllers/shopping_lists_controller.rb

```diff
@@ -0,0 +1,18 @@
+class ShoppingListsController < ApplicationController
+  def index
+    @recipe = Recipe.find(params[:recipe_id])
+    @recipe_foods = @recipe.recipe_foods.includes(:food).order(:id)
+
+    @missing_ingredients = @recipe_foods.select { |ingredient| ingredient.quantity > ingredient.food.quantity }
+    @missing_ingredients = @missing_ingredients.map do |ingredient|
+      {
+        name: ingredient.food.name,
+        unit: ingredient.food.measurement_unit,
+        quantity: ingredient.quantity - ingredient.food.quantity,
+        price: ingredient.food.price * (ingredient.quantity - ingredient.food.quantity)
+      }
+    end
+
+    @total_price = @missing_ingredients.map { |ingredient| ingredient[:price] }.sum
+  end
+end
```

Comment:
### Your Shopping List URL
![image](https://user-images.githubusercontent.com/93039359/219693356-646ee03e-5df6-49c1-9b2c-31a584906b64.png)

### Wireframe Shopping List URL
![image](https://user-images.githubusercontent.com/93039359/219693242-02199dcf-822e-47f2-985c-61e061ed577b.png)

- [ ] It would be nice if your URL would match with the wireframe URL :+1: here this requirement is so much important because of security reasons. You would not want to disclose your users personal information like their id's to other people :+1: so kindly fix it :blush:

Reply:
Route updated

Reply:
resolved! ✔️